js/src/pages/ProjectRolePage.tsx

```diff
@@ -56,27 +54,28 @@ export const ProjectRolePage: React.FC = () => {
     [dispatch],
   );
 
-  const errors = useSelector(getErrors);
-  const loading = useSelector(getIsLoading);
+  const fetchRolePermissions = useCallback(() => {
+    dispatch(fetchProjectRoleAsync.request());
+  }, [dispatch]);
+
   const project = useSelector(getProject);
+  const rolePermissions = useSelector(getRolePermissions);
 
   useEffect(() => {
     fetchProject(slug);
-  }, [location, slug, fetchProject]);
-
-  useClearState(clearStateAction);
+    fetchRolePermissions();
+  }, [location, slug, fetchProject, fetchRolePermissions]);
```

Comment:
why you use location and slug together? If location changes slug will change as well.  I think you can use only one `slug` without 'location'.

Reply:
done